sound/soc/sof/compress.c

```diff
@@ -209,6 +212,9 @@ static int sof_compr_set_params(struct snd_soc_component *component,
 	pcm->params.sample_container_bytes =
 		snd_pcm_format_physical_width(SNDRV_PCM_FORMAT_S32) >> 3;
 	pcm->params.host_period_bytes = params->buffer.fragment_size;
+	pcm->params.ext_data_length = data_size;
+
+	memcpy((u8*)pcm->params.data, &params->codec, data_size);
```

Comment:
Shouldn't we have something to check the data_size is compatible with the IPC size requirements?

Reply:
good point. I've added a check above.